Dockerfile

```
@@ -1,5 +1,35 @@
+FROM node:14.14.0 as node-build
+
+ADD . /hydroshare
+
+WORKDIR /hydroshare/hs_discover
+
+RUN rm -rf static templates && \
+    mkdir static templates && \
+    mkdir templates/hs_discover && \
+    mkdir static/js && \
+    mkdir static/css && \
+    npm install && \
+    npm run build && \
+    mkdir -p static/js && \
+    mkdir -p static/css && \
+    cp -rp templates/hs_discover/js static/ && \
+    cp -rp templates/hs_discover/css static/ && \
+    cp -p templates/hs_discover/map.js static/js/ && \
+    echo "----------------js--------------------" && \
+    ls -l static/js && \
+    echo "--------------------------------------" && \
+    echo "----------------css-------------------" && \
+    ls -l static/css && \
+    echo "--------------------------------------" && \
+    cd static/ && \
+    cp js/app.*.js js/app.js && \
+    cp js/chunk-vendors.*.js js/chunk-vendors.js
```

The hash that the Vue CLI build creates in the `js/app.*.js` file (and same for the chunk-vendors file) is important for versioning between builds (ex for cache busting). I don't know why we overwrote them historically in hsctl but I think we might be better off preserving them? Perhaps something like:

```suggestion
    cp templates/hs_discover/css/* static/css/ && \
    cp templates/hs_discover/js/* static/js/ && \
    cp templates/hs_discover/map.js static/js/map.js && \
    echo "----------------js--------------------" && \
    ls -l static/js && \
    echo "--------------------------------------" && \
    echo "----------------css-------------------" && \
    ls -l static/css && \
    echo "--------------------------------------" && \
```

---

disregard.
I forgot that we are using [ManifestStaticFilesStorage](https://github.com/hydroshare/hydroshare/blob/node-to-multistage-docker-build/hydroshare/storage.py#L5) for our static files in django